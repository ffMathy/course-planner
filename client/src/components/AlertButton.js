import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogContent,
    AlertDialogOverlay,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    IconButton,
} from "@chakra-ui/core";

const AlertButton = ({ itemType, itemName, action, navigateTo, confirmFn }) => {
    const [isOpen, setIsOpen] = React.useState();
    const onClose = () => setIsOpen(false);
    const cancelRef = React.useRef();

    return (
        <>
            <IconButton icon="delete" variantColor="red" size="lg" onClick={() => setIsOpen(true)} />
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button
                            variantColor="red"
                            onClick={() => {
                                onClose();
                                confirmFn();
                            }}
                            ml={3}
                        >
                            Delete
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default AlertButton;
