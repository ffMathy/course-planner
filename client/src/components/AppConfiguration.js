import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={customTheme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

// See: https://chakra-ui.com/getting-started
const customTheme = {
    ...theme,
};

export default AppConfiguration;
