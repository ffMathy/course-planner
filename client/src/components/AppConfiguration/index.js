import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={theme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

export default AppConfiguration;
