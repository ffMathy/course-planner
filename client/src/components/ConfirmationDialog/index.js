import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogOverlay,
    AlertDialogContent,
    AlertDialogHeader,
    AlertDialogBody,
    AlertDialogFooter,
} from "@chakra-ui/core";

const ConfirmationDialog = ({ isOpen, onClose, confirm, itemType, itemName, action, navigateTo }) => {
    const cancelRef = React.useRef();

    return (
        <>
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button variantColor="red" onClick={confirm} ml={3}>
                            {action}
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default ConfirmationDialog;
