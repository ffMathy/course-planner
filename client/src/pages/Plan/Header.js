import React from "react";
import { Flex, Text } from "@chakra-ui/core";
import { colors as c } from "../../colors";

const Header = ({ name, programme, planName }) => {
    return (
        <Flex direction="row" p={4} justifyContent="space-around">
            <Flex direction="column">
                <Flex align="center" justify="center">
                    <Text color={c.uoaBlue} fontSize="xl">
                        Student
                    </Text>
                </Flex>
                <Flex
                    minWidth="100%"
                    height="50px"
                    backgroundColor={c.uoaBlue}
                    borderRadius="10px"
                    align="center"
                    justify="center"
                    p={5}
                    marginTop={2}
                >
                    <Text textAlign="center" color={c.white} fontSize="xl">
                        {name}
                    </Text>
                </Flex>
            </Flex>

            <Flex align="center" justify="center">
                <Text color={c.uoaBlue} fontSize="5xl" fontWeight="bold">
                    {planName}
                </Text>
            </Flex>

            <Flex direction="column">
                <Flex align="center" justify="center">
                    <Text color={c.uoaBlue} fontSize="xl">
                        Program
                    </Text>
                </Flex>
                <Flex
                    minWidth="100%"
                    height="50px"
                    backgroundColor={c.uoaBlue}
                    borderRadius="10px"
                    align="center"
                    justify="center"
                    p={5}
                    marginTop={2}
                >
                    <Text textAlign="center" color={c.white} fontSize="xl">
                        {programme}
                    </Text>
                </Flex>
            </Flex>
        </Flex>
    );
};

export default Header;
