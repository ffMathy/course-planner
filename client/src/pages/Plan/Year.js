import React, { useState } from "react";
import { Flex, Text, Stack } from "@chakra-ui/core";
import { useDrop } from "react-dnd";
import { ItemTypes } from "./ItemTypes";
import { colors as c } from "../../colors";

const CourseTile = ({ courseName }) => {
    return (
        <Flex p={2} justify="center" align="center" borderWidth="1px" width="90%" backgroundColor={c.darkGrey} borderRadius="5px" mt="2px">
            <Text textAlign="center" fontWeight="bold" fontSize="xl" color={c.white}>
                {courseName}
            </Text>
        </Flex>
    );
};

const SemesterBox = ({ semester }) => {
    const [courses, setCourses] = useState(["COMPSYS201", "ENGSCI213", "SOFTENG250"]);

    const [{ canDrop, isOver }, drop] = useDrop({
        accept: ItemTypes.COURSE_PILL,
        drop: ({ courseName }, monitor) => {
            setCourses([...courses, courseName]);
            return { name: "Dustbin" };
        },
        collect: (monitor) => ({
            isOver: monitor.isOver(),
            canDrop: monitor.canDrop(),
        }),
    });

    const isActive = canDrop && isOver;
    let backgroundColor = "#E2E2E2";
    if (isActive) {
        backgroundColor = "#A9A9A9";
    } else if (canDrop) {
        backgroundColor = "#E2E2E2";
    }

    return (
        <Flex direction="column" width="50%" justify="center" align="center">
            <Flex width="100%" justify="center" align="center" marginTop="20px">
                <Text textAlign="center" fontWeight="bold" fontSize="xl" color={c.uoaBlue}>
                    {semester}
                </Text>
            </Flex>
            <Flex
                width="75%"
                justify="center"
                align="center"
                marginTop="5px"
                backgroundColor={c.lightGrey}
                paddingTop="2"
                paddingBottom="2"
                style={{ backgroundColor }}
            >
                <Stack className="courseContainer" spacing={8} width="100%" align="center" maxHeight="200px" overflowY="scroll" ref={drop}>
                    {courses.map((course, idx) => (
                        <CourseTile key={idx} courseName={course} />
                    ))}
                </Stack>
            </Flex>
        </Flex>
    );
};

const Year = ({ year }) => {
    return (
        <Flex direction="column" mt={4}>
            <Flex width="100%" justify="center" align="center" marginTop="20px">
                <Text textAlign="center" fontWeight="bold" fontSize="3xl" color={c.darkGrey}>
                    {year}
                </Text>
            </Flex>
            <Flex width="100%" direction="row" marginTop="5px">
                <SemesterBox semester={"Semester One"} />
                <SemesterBox semester={"Semester Two"} />
            </Flex>
        </Flex>
    );
};

export default Year;
