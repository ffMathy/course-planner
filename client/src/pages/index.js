import HomePage from "./HomePage";
import CoursesPage from "./CoursesPage";
import ProgrammesPage from "./Programmes/ProgrammesPage";
import Students from "./Students";

export { HomePage, CoursesPage, ProgrammesPage, Students };
