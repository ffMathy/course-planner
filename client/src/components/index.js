import AppConfiguration from "./AppConfiguration";
import OutlineButton from "./OutlineButton";
import ProgrammeCard from "./ProgrammeCard";
import SearchBar from "./SearchBar";
import ProgrammeRequirementsItem from "./ProgrammeRequirementsItem";
import ConfirmationDialog from "./ConfirmationDialog";

export { AppConfiguration, OutlineButton, ProgrammeCard, SearchBar, ProgrammeRequirementsItem, ConfirmationDialog };
