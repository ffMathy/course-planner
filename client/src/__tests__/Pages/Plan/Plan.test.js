import React from "react";
import renderer from "react-test-renderer";
import { AppConfiguration } from "../../../components";
import { Plan } from "../../../pages";
import Header from "../../../pages/Plan/Header";
import Year from "../../../pages/Plan/Year";

test("<Plan />", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Plan />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<Header />", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Header name={"Nisarag"} programme={"SOFTENG 701"} lastSaveDate={new Date()} planName={"HongShi"} />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
