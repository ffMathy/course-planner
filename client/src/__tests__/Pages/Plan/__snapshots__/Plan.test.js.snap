// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plan /> 1`] = `
<div
  className="css-m5hsxh"
>
  <div
    className="css-lbicpo"
  >
    <div
      className="css-k008qs"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          className="css-1tqsd30"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          type="button"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
          />
        </svg>
      </a>
    </div>
    <div
      className="css-u6v8er"
    >
      <div
        className="css-u4p24i"
      >
        <p
          className="css-1g1e9sz"
        >
          Requirements
        </p>
        <svg
          className="css-rw24u3"
          focusable={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm0,19a1.5,1.5,0,1,1,1.5-1.5A1.5,1.5,0,0,1,12,19Zm1.6-6.08a1,1,0,0,0-.6.917,1,1,0,1,1-2,0,3,3,0,0,1,1.8-2.75A2,2,0,1,0,10,9.255a1,1,0,1,1-2,0,4,4,0,1,1,5.6,3.666Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="css-1lads1q"
      >
        <div
          className="css-51am61"
        >
          <div
            className="css-bjsx6g"
          >
            <p
              className="css-1t07m2d"
            >
              1
              .
            </p>
            <p
              className="css-3n7v07"
            >
              EXACTLY
            </p>
            <p
              className="css-12kv7x2"
            >
              40 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            >
              <svg
                className="css-1im46kq"
                focusable={false}
                role="presentation"
                viewBox="0 0 14 14"
              >
                <g
                  fill="currentColor"
                >
                  <polygon
                    points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="css-13y1gzu"
          >
            <p
              className="css-1t07m2d"
            >
              2
              .
            </p>
            <p
              className="css-3n7v07"
            >
              ATLEAST
            </p>
            <p
              className="css-12kv7x2"
            >
              0 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            />
          </div>
          <div
            className="css-13y1gzu"
          >
            <p
              className="css-1t07m2d"
            >
              3
              .
            </p>
            <p
              className="css-3n7v07"
            >
              EXACTLY
            </p>
            <p
              className="css-12kv7x2"
            >
              15 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            />
          </div>
          <div
            className="css-bjsx6g"
          >
            <p
              className="css-1t07m2d"
            >
              4
              .
            </p>
            <p
              className="css-3n7v07"
            >
              ATLEAST
            </p>
            <p
              className="css-12kv7x2"
            >
              30 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            >
              <svg
                className="css-1im46kq"
                focusable={false}
                role="presentation"
                viewBox="0 0 14 14"
              >
                <g
                  fill="currentColor"
                >
                  <polygon
                    points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="css-13y1gzu"
          >
            <p
              className="css-1t07m2d"
            >
              5
              .
            </p>
            <p
              className="css-3n7v07"
            >
              EXACTLY
            </p>
            <p
              className="css-12kv7x2"
            >
              15 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            />
          </div>
          <div
            className="css-13y1gzu"
          >
            <p
              className="css-1t07m2d"
            >
              6
              .
            </p>
            <p
              className="css-3n7v07"
            >
              UPTO
            </p>
            <p
              className="css-12kv7x2"
            >
              40 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-chxm8w"
            >
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
              <p
                className="css-1usq9ww"
              >
                SOFTENG370
              </p>
            </div>
            <div
              className="css-k008qs"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1ofqig9"
    >
      <p
        className="css-1g1e9sz"
      >
        Courses
      </p>
      <div
        className="css-9jn3o8"
      >
        <div
          className="css-yh0f2e"
        >
          <div
            className="css-1lads1q"
          >
            <div
              className="css-aetr73"
            >
              <div
                className="css-k6882c"
              >
                <svg
                  className="css-1xb1rov"
                  focusable={false}
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <input
                className="css-n5gi9c"
                onChange={[Function]}
                placeholder="Search Courses"
                type="search"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-1a105je"
        />
      </div>
      <p
        className="css-1g1e9sz"
      >
        Notes
      </p>
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-w80loo"
        >
          <textarea
            className="css-ro0llr"
            onChange={[Function]}
            placeholder="Add your note here"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-tq7xf1"
  >
    <div
      className="css-1sqeryc"
    >
      <div
        className="css-j7qwjs"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="css-1gwygh1"
          >
            Student
          </p>
        </div>
        <div
          className="css-5l6ak"
        >
          <p
            className="css-erfz49"
          />
        </div>
      </div>
      <div
        className="css-gmuwbf"
      >
        <p
          className="css-137ozra"
        />
      </div>
      <div
        className="css-j7qwjs"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="css-1gwygh1"
          >
            Program
          </p>
        </div>
        <div
          className="css-5l6ak"
        >
          <p
            className="css-erfz49"
          />
        </div>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      className="css-1gi9ej6"
    />
    <div
      className="css-1apjpfw"
    />
  </div>
</div>
`;
